docs/future/migration.md

```diff
@@ -0,0 +1,94 @@
+# Migration Guide (for React 0.13 - React 15.x)
+
+
+## Root Wrapper
+
+The initially returned wrapper used to be around the element passed
+into the `mount` API, and for `shallow` it was around the root node of the rendered output of the element passed in. After the upgrade, the
+two APIs are now symmetrical, starting off
+
+```js
+const x = 'x';
+const Foo = props => <div inner={props.outer} />
+const wrapper = mount(<Foo outer={x} />);
+```
+
+```js
+expect(wrapper.props()).to.deep.equal({ outer: x });
+```
+
+## Refs
+
+Refs no longer return a "wrapper". They return what the ref would actually be.
+
+
+## Keys
+
+keys no longer work? we should maybe fix this in the spec...
+
+
+## for shallow, getNode() was renamed to getElement()
+
+## for mount, getNode() should not be used. instance() does what it used to.
+
+## for mount, getElement() will return the root JSX element
+
+## what getNode() returns
+
+we need to keep in mind that `getElement()` will no longer be referentially equal to what it was before.
+
+## Updates are required
+
+```
+wrapper.find('.async-btn').simulate('click');
+setImmediate(() => {
+  // this didn't used to be needed
+  wrapper.update(); // TODO(lmr): this is a breaking change...
```

Comment:
yes, the new architecture makes this fundamentally not possible. I don't see any way around this.